internal/static/testdata/pkg-with-errors/app01/app.go

```diff
@@ -0,0 +1,22 @@
+package app01
```

Comment:
I like the naming of the directories that are directly below `testdata` because they are descriptive. It would be good to keep that up with the package and typenames below that (rather than numbered packages).

Reply:
[This commit](https://github.com/dogmatiq/configkit/pull/118/commits/79bfd311c5667c114ca84404d70cd7e661e1b7dd) restructures tests and make the `testdata` subdirectories more descriptive. I also went a bit further and split the test files to separate the testing of Dogma application detection logic and Dogma application identity population logic. Please take a look when you have time and LMK if this is an appropriate structure.